lib/core/src/Cardano/Wallet/Primitive/CoinSelection/MA/RoundRobin.hs

```diff
@@ -756,8 +756,29 @@ makeChange
         totalMinCoinValue =
             F.sum $ (coinToNatural . minCoinValueFor) <$> change
 
+    -- Outputs tokens are ordered in such way that to the greatest extent
+    -- possible, small (resp. large) quantities of non user-defined change assets
+    -- will be merged with small (resp. large) quantities of user-defined change
+    -- assets.
+    --
+    -- There's no simple way to define a total order on 'TokenMap', so we opt
+    -- for the following approach:
+    --
+    --     - If there exists a partial order as defined in the TokenMap module,
+    --       we use it.
+    --
+    --     - Otherwise, we order them using the maximum quantity of one of their
+    --       assets.
+    --
+    -- Over time, this should tend to bundle small (resp. large) bundles together.
     outputTokens :: NonEmpty TokenMap
-    outputTokens = view #tokens <$> outputBundles
+    outputTokens = NE.sortBy totalOrder (view #tokens <$> outputBundles)
+      where
+        totalOrder m1 m2
+            | m1 `leq` m2 = LT
```

Comment:
I think this is not _quite_ right, as `leq` will also return `True` if two maps are equal.

To fix it, we'd need to introduce an additional guard:
```hs
             | m1 ==    m2 = EQ
             | m1 `leq` m2 = LT
```

However, this is unfortunately still not enough to generate a well-defined total ordering, as even the modified function fails when tested with `testLaws`.

However, we can make a total ordering where we first compare on the greatest token quantity, and then fall back to lexicographic ordering in the event of a tie-break. I made a patch that tests this here: 185a712b41c91b2203685044b2f9d246efd2fcb4